### PR TITLE
PageFormExecutionEnvironmentSelect Update

### PR DESCRIPTION
**Pull request by jamestalton**
Updated `PageFormExecutionEnvironmentSelect` to use the `PageFormSingleSelectAwxResource` component.

Our select dropdowns current have a bug where if an update happens, the browse dialog will go away. PageFormSingleSelectAwxResource fixes that bug and also updates the dropdown to use the newer PageSingleSelect.

### CODE REVIEW

File: framework/PageInputs/PageSingleSelect.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import {
   SelectList,
```
Comment (MilanPospisil) [inline framework/PageInputs/PageSingleSelect.tsx:12]: Forgotten changes for PageMultiSelect?

**Comment by MilanPospisil** [comment]
Are we going to do those extensions also in multi versions?

File: framework/PageInputs/PageSingleSelect.tsx [hunk 2]
```diff
@@ -162,42 +162,49 @@ export function PageSingleSelect<
 
   const selectedLabel = useMemo(() => {
     let selectedLabel: ReactNode = selectedOption?.label;
-    if (!selectedLabel && value !== undefined && value !== '') {
+    if (!selectedLabel && value !== undefined && value !== '' && value !== null) {
```
Comment (MilanPospisil) [inline framework/PageInputs/PageSingleSelect.tsx:165]: what about simplification:
!selectedLabel && value

Comment (jamestalton) [reply framework/PageInputs/PageSingleSelect.tsx:165]: If value was 0 that would fail.

**Comment by jamestalton** [comment]
> Are we going to do those extensions also in multi versions?

Yes

File: framework/PageInputs/PageSingleSelect.tsx (hunk 2 above)
Comment (MilanPospisil) [reply framework/PageInputs/PageSingleSelect.tsx:165]: Oh, so it is not only string, makes sense.

**Review by akus062381 — CHANGES_REQUESTED** [review]
Can we give a user the ability to remove a selected form field option for a form field that is not marked as required?

**Comment by akus062381** [comment]
When I go to the Execution Environment form field and select Browse, I see the following:

image

**Comment by akus062381** [comment]
When I click into the Edit form for the Job Template, it shows the execution environment's ID instead of its name. Can we change this so the name is displayed?
image

**Pull request closed without merging**